libs/barista-components/slider/src/slider.ts

```diff
@@ -0,0 +1,401 @@
+    }
+  }
+  /** Holds the value of step internally */
+  private _step: number = 5;
+
+  /** TODO: implement the usage of this binding */
+  @Input() disabled: boolean = false;
+
+  private _value: number;
+
+  @Input()
+  get value(): number {
+    return this._value;
+  }
+  set value(value: number) {
+    this._updateValue(value);
+  }
+
+  /** Updates the value if the update is triggered by the consumer. */
+  private _updateValue(value: number, userTriggered: boolean = true): void {
+    this._value = value;
+    // We only need to update if the update is coming from outside the component.
+    if (userTriggered) {
+      this._value$.next(roundToSnap(value, this.step, this.min, this.max));
+    }
+  }
+
+  /**
+   * Convert input string value to number and call
+   * roundToSnap takes care of snapping the values to the steps
+   */
+  inputValueChanged(event: Event): void {
+    this.inputFieldValue$.next(
+      +(event.currentTarget as HTMLInputElement).value,
+    );
+  }
+
+  /** Provides event for value change */
+  @Output() change = new EventEmitter<number>();
+
+  /** @internal Holds the track wrapper */
+  @ViewChild('trackWrapper', { static: true })
+  _trackWrapper: ElementRef<HTMLDivElement>;
+
+  /** @internal Holds the thumb */
+  @ViewChild('thumb', { static: true })
+  _thumb: ElementRef<HTMLDivElement>;
+
+  /** @internal Holds the slider fill */
+  @ViewChild('sliderFill', { static: true })
+  _sliderFill: ElementRef<HTMLDivElement>;
+
+  /** @internal Holds the slider background */
+  @ViewChild('sliderBackground', { static: true })
+  _sliderBackground: ElementRef<HTMLDivElement>;
+
+  /** @internal Holds the slider input field */
+  @ViewChild(DtInput, { static: true })
+  _sliderInput: DtInput;
+
+  /** Observer that completes on ngOnDestroy */
+  private _destroy$ = new Subject<void>();
+
+  constructor(private _zone: NgZone, private _platform: Platform) {}
+
+  ngOnInit(): void {
+    const initValue = roundToSnap(this.value, this.step, this.min, this.max);
+    if (this._value$.value !== initValue) {
+      this._updateValue(initValue);
+    }
+  }
+
+  ngAfterViewInit(): void {
+    if (this._platform.isBrowser && 'ResizeObserver' in window) {
+      this._observer = new window.ResizeObserver(_ => {
+        this._resizeObserver$.next();
+      });
+
+      this._observer.observe(this._trackWrapper.nativeElement);
+    }
+
+    this._clientRect$ = merge(
+      this._resizeObserver$.pipe(debounceTime(50)),
+      of(null), // at least one initial trigger is needed
+    ).pipe(
+      map(() => this._trackWrapper.nativeElement.getBoundingClientRect()),
+      takeUntil(this._destroy$),
+    );
+
+    this._zone.runOutsideAngular(() => {
+      this._captureEvents();
+    });
+  }
+
+  ngOnDestroy(): void {
+    this._destroy$.next();
+    this._destroy$.complete();
+    if (this._platform.isBrowser && 'ResizeObserver' in window) {
+      this._observer.unobserve(this._trackWrapper.nativeElement);
+    }
```

Comment:
```suggestion
    if (this._platform.isBrowser && this._observer) {
      this._observer.disconnect();
    }
```
You don't really care about, if the resizeObserver is available in the window or not. In this case we care if it is set. 
And it is only set, if it was initialized in the ngAfterViewInit.

When calling disconnect, it will disconnect from all observed elements. This seems safer than unobserving a single native element on destroy.

Reply:
Yepp, you are right, this seems more reasonable.